### ISSUE TITLE
[MODEL] MTS-LSTM (Multi-Timescale Long Short-Term Memory Network)

### ISSUE THREAD

**Issue by joannawang-100** [Flood-Lab/HydroTuring#126]
### Proposer

Jiao Wang, Department of Land, Air and Water Resources, University of California Davis, California, USA. orcid: 0000-0002-4029-5814

### Model name

 MTS-LSTM

### Model category

AI-based

### Model code repository

https://github.com/gauchm/mts-lstm The model implementation is provided by NeuralHydrology: https://github.com/neuralhydrology/neuralhydrology

### Version, release, or commit

MTS-LSTM v1.0, using NeuralHydrology commit 9626578. Archived release: https://doi.org/10.5281/zenodo.4687992

### Paper or model reference (optional)

https://doi.org/10.5194/hess-25-2045-2021 — Gauch et al. (2021), Rainfall–runoff prediction at multiple timescales with a single Long Short-Term Memory network, Hydrology and Earth System Sciences, 25, 2045–2062.

### Code and weights licenses

Code: BSD-3-Clause for the NeuralHydrology model implementation. Weights: CC BY 4.0 through Zenodo.

### Supported timestep

PT1H (hourly)

### Required inputs

Forcing: hourly and daily precipitation (mm/h and mm/day), air temperature (°C), radiation (W/m²), specific humidity (kg/kg), atmospheric pressure (Pa), and wind speed (m/s).
Static: catchment area (km²) and CAMELS catchment attributes.
History: 365 daily time steps and 336 hourly time steps.
Initialization: a pretrained regional checkpoint.

### Outputs produced by the model

dis — river discharge

### Packaging status

Help is needed with both Docker packaging and the adapter

### Installation and inference notes

MTS-LSTM is implemented in Python through the NeuralHydrology package. The archived experiment environment can be created from environment.yml, followed by installing the paper's pinned NeuralHydrology revision:

`pip install https://github.com/neuralhydrology/neuralhydrology/archive/9626578.zip`

Inference is run with: `nh-run evaluate --run-dir <run_directory> --period test`

Published checkpoints, configurations, and normalization data are available from Zenodo. The full MTS-LSTM archive is approximately 21.8 GB because it contains multiple experiments and ten random seeds. A HydroTuring container should retain one documented checkpoint and its required configuration rather than package the full archive.

The model can run with PyTorch on CPU, although GPU inference would be faster. All dependencies, checkpoint files, normalization statistics, and required catchment attributes must be included in the container for offline evaluation.

### Why this model belongs in the benchmark

MTS-LSTM predicts discharge at hourly and daily resolutions within one network and was designed to improve consistency across temporal scales. This creates a direct, testable question for HydroTuring: whether hourly discharge aggregated over a day agrees with the model's independently generated daily discharge.

Unlike a conventional LSTM evaluated at one timestep, MTS-LSTM can reveal whether apparently accurate predictions remain hydrologically consistent when the same event is represented at different temporal resolutions. It therefore provides a strong AI test case for resolution invariance and cross-timescale water-volume consistency.

### Submission confirmation

- [x] I have identified a reproducible code revision and have permission to use the linked code and weights for this evaluation.
- [x] I understand that HydroTuring may report PASS, FAIL or N/A on each probe, with a reason of VIOLATION, INCOMPLETE, INCOMPATIBLE or ERROR.
